### PR TITLE
chore: add empty state screens and implement filtering logic [part 1]

### PR DESCRIPTION
**Pull request by MounirDhahri**
This PR resolves [CX-3638] 

### Description

This PR makes the following changes:
- Update image for empty state screen
- Implement (has no artworks + has artists collected)
- Add pills filtering logic (was needed to finish the work on the filtering)
- Extract the empty screen into a separate component

https://user-images.githubusercontent.com/11945712/235911149-3f8c9106-c183-49f6-9ba4-12c81dd64e48.mov

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Add empty state to collected artists/artworks - mounir
- Implement pill filtering logic inside my collection - mounir

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[CX-3638]: https://artsyproduct.atlassian.net/browse/CX-3638?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/MyCollection/Components/MyCollectionStickyHeader.tsx [hunk 1]
```diff
@@ -218,7 +215,7 @@ const AnimatedPill = ({
   }
 
   const selected = selectedTab === tab
-  const disabled = !!selectedTab && selectedTab !== tab
+  const pressable = (!!selectedTab && selectedTab !== tab) || disabled
```
Comment (olerichter00) [inline src/app/Scenes/MyCollection/Components/MyCollectionStickyHeader.tsx:218]: **suggestion:** Shouldn't this be the opposite? Right now the pill is pressable if it's disabled 🤔

File: src/app/Scenes/MyCollection/Components/MyCollectionZeroStates.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
+import { ActionType, AddCollectedArtwork, ContextModule, OwnerType } from "@artsy/cohesion"
+import { useSpace, Flex, LockIcon, Button, Text } from "@artsy/palette-mobile"
+import { ZeroState } from "app/Components/States/ZeroState"
+import { StickyTabPageScrollView } from "app/Components/StickyTabPage/StickyTabPageScrollView"
+import { Tab } from "app/Scenes/MyProfile/MyProfileHeaderMyCollectionAndSavedWorks"
+import { navigate, popToRoot } from "app/system/navigation/navigate"
+import { Image } from "react-native"
+import { useTracking } from "react-tracking"
+
+export const MyCollectionZeroState: React.FC = () => {
```
Comment (olerichter00) [inline src/app/Scenes/MyCollection/Components/MyCollectionZeroStates.tsx:10]: **suggestion:** In my opinion it's better to export one component per file and name the file like the component to be able to make searching for components easier.

**Review by olerichter00 — APPROVED** [review]
Great work 🌟

File: src/app/Scenes/MyCollection/Components/MyCollectionStickyHeader.tsx (hunk 1 above)
Comment (MounirDhahri) [reply src/app/Scenes/MyCollection/Components/MyCollectionStickyHeader.tsx:218]: trueee 😄 do you have a name you suggest

File: src/app/Scenes/MyCollection/Components/MyCollectionZeroStates.tsx (hunk 2 above)
Comment (MounirDhahri) [reply src/app/Scenes/MyCollection/Components/MyCollectionZeroStates.tsx:10]: that's valid feedback - I was thinking of this more as a helper than a component but I understand where your comment comes from. I will gladly update it given it's a small update

File: src/app/Scenes/MyCollection/Components/MyCollectionStickyHeader.tsx (hunk 1 above)
Comment (olerichter00) [reply src/app/Scenes/MyCollection/Components/MyCollectionStickyHeader.tsx:218]: Hmm, wither passing `!!selectedTab && selectedTab !== tab` directly as a prop or something like:

```suggestion
  const disabled = (!!selectedTab && selectedTab !== tab) || disabledProp
```

Comment (MounirDhahri) [reply src/app/Scenes/MyCollection/Components/MyCollectionStickyHeader.tsx:218]: good suggestion - i will change

**Pull request merged**